video/cloud-client/analyze/beta_snippets.py

```diff
@@ -0,0 +1,207 @@
+# [END video_face_bounding_boxes]
+
+
+# [START video_face_emotions]
+def face_emotions(path):
+    """ Analyze faces' emotions over frames. """
+    video_client = videointelligence.VideoIntelligenceServiceClient()
+    features = [videointelligence.enums.Feature.FACE_DETECTION]
+
+    # include_bounding_boxes must be set to True for include_emotions
+    # to work.
+    config = videointelligence.types.FaceConfig(
+        include_bounding_boxes=True,
+        include_emotions=True)
+    context = videointelligence.types.VideoContext(
+        face_detection_config=config)
+
+    operation = video_client.annotate_video(
+        path, features=features, video_context=context)
+    print('\nProcessing video for face annotations:')
+
+    result = operation.result(timeout=600)
+    print('\nFinished processing.')
+
+    # There is only one result because a single video was processed.
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        frame_emotions = []
+        for frame in face.frames:
+            time_offset = (frame.time_offset.seconds +
+                           frame.time_offset.nanos / 1e9)
+            emotions = frame.attributes[0].emotions
+
+            # from videointelligence.enums
+            emotion_labels = (
+                'EMOTION_UNSPECIFIED', 'AMUSEMENT', 'ANGER',
+                'CONCENTRATION', 'CONTENTMENT', 'DESIRE',
+                'DISAPPOINTMENT', 'DISGUST', 'ELATION',
+                'EMBARRASSMENT', 'INTEREST', 'PRIDE', 'SADNESS',
+                'SURPRISE')
+
+            emotion, score = sorted(
+                [(em.emotion, em.score) for em in emotions],
+                key=lambda p: p[1])[-1]
+            emotion_label = emotion_labels[emotion]
+
+            frame_emotions.append((time_offset, emotion_label, score))
+
+        frame_emotions = sorted(frame_emotions, key=lambda p: p[0])
+        for time_offset, emotion_label, score in frame_emotions:
+            print('\t{:04.2f}s: {:14}({:4.3f})'.format(
+                time_offset, emotion_label, score))
+        print('\n')
+# [END video_face_emotions]
+
+
+# [START video_speech_transcription]
+def speech_transcription(input_uri):
+    """Transcribe speech from a video stored on GCS."""
+    video_client = videointelligence.VideoIntelligenceServiceClient()
+
+    features = [
+        videointelligence.enums.Feature.SPEECH_TRANSCRIPTION
+    ]
+
+    config = videointelligence.types.SpeechTranscriptionConfig(
+        language_code='en-US')
+    video_context = videointelligence.types.VideoContext(
+        speech_transcription_config=config)
+
+    operation = video_client.annotate_video(
+        input_uri, features=features,
+        video_context=video_context)
+
+    print('\nProcessing video for speech transcription.')
+
+    result = operation.result(timeout=180)
+
+    annotation_results = result.annotation_results[0]
```

Comment:
Can you add a comment that notes you are only pulling out the first result or is there only one result?

Reply:
done.